Dat 3732 - collapsible sections

rybmat opened the pull request:
handling collapsible section on the app side of Infobox Builder

rybmat committed:
support for collapse attrib in group node while converting xml->json

rybmat committed:
support for collapse=open in group node while converting from json to xml

rybmat committed:
cleanup

rybmat committed:
cleanup

rybmat committed:
extracting logic creating group dom to one method

Commit:
DAT-3732, move collpsible group chevron styles to mixin and use it in preview

sebastianmarzjan commented on extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:
```diff
@@ -16,7 +16,26 @@ class NodeGroup extends Node {
+	/**
+	 * @var array of string
+	 */
+	protected $allowedAttributes = [ 'collapse' ];
```
We need to have check for supported (allowed) values of this attribute too (also in the tests)

rybmat replied:
done

sebastianmarzjan commented on extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
 	}
 
 	/**
+	 * @param $collapse
```
This seems off

sebastianmarzjan commented on extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
-		return dom_import_simplexml(
-			new SimpleXMLElement( '<' . \Wikia\PortableInfoboxBuilder\Nodes\NodeGroup::XML_TAG_NAME . '/>' )
-		);
+	protected function createGroupDom($childHeader, $childNodeDom) {
```
hmm, what do you think about

```
protected function createGroupDom($headerNodeDom, $collapsible) { ... }
```

invoked in 
https://github.com/Wikia/app/pull/9819/files#diff-f997a9b22af5a0ef703dfc816b8678ebR202 
and
https://github.com/Wikia/app/pull/9819/files#diff-f997a9b22af5a0ef703dfc816b8678ebR216
like

```
$currentGroupDom = $this->createGroupDom( $childNodeDom, $this->isCollapsibleHeader($currentChildNode));
```

That would make the createGroupDom more explicit and ensure the share of responsibilities.

sebastianmarzjan commented on extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
+		}
+
+		// clean the mess connected to differences in xml tree and data model on mercury side
+		$childNodeDom->removeAttribute('section_collapsible');
```
Please create a new header instead or modify it in the generic flow of getFormattedMarkup in which it is created, as we discussed offline.

rybmat committed:
checking if group doesn't have collapse attr other than 'open'

rybmat committed:
refactor of handling collapsible sections in json->xml translation

rybmat committed:
Merge branch 'DAT-3732' of github.com:Wikia/app into DAT-3732

rybmat committed:
messages in markup support test

rybmat committed:
comments, casting to bool

sebastianmarzjan commented:
Waiting for tests, and LGTM if they pass

Commit:
Merge branch 'dev' into DAT-3732

Commit:
DAT-3732, fix whitespace issues

dianafa committed:
add space

Commit:
Fix tests failing after introduction of strict comparison ( '===' instead of '==' ).
In these cases, automatic type cast does not happen and one needs to cast the SimpleXML resource
to string explicitly.

Commit:
DAT-3732, support empty headers

Commit:
DAT-3732, fix issue with empty header height

Commit:
DAT-3732, add test for empty header
